pkg/datastore/headerscontroller.go

```diff
@@ -18,17 +18,20 @@ type HeaderController struct {
 
 	latestSlot *uint64
 	cl         sync.RWMutex
+
+	m HeaderControllerMetrics
 }
 
 func NewHeaderController(slotLag uint64, slotTimeLag time.Duration) *HeaderController {
 	latestSlot := uint64(0)
-	return &HeaderController{
+	hc := &HeaderController{
 		slotLag:     slotLag,
 		slotTimeLag: slotTimeLag,
-
-		latestSlot: &latestSlot,
-		headers:    make(map[uint64]*IndexedHeaders),
+		latestSlot:  &latestSlot,
+		headers:     make(map[uint64]*IndexedHeaders),
 	}
+	hc.initMetrics()
```

Comment:
the initMetrics was not called before. Why do we call it no?

Reply:
The other ones are static, it's initiated as in the other ones